src/fluids.md

```diff
@@ -0,0 +1,184 @@
+    \frac{b_0}{\Delta t} u^{n+1} = -\nabla p^{n+1} + L(u^{n+1}) + F^\*(u^n).
+\end{align}
+
+To achieve a high order convergence in space the linear term $L(u)$ is replaced
+by
+
+\begin{align}
+    L_{\times}(u) = \nu \nabla(\nabla \cdot u) - \nu \nabla \times \nabla \times u
+\end{align}
+
+which is used to weakly enforce incompressibility by setting the first term to
+zero.
+
+To compute the pressure we rearrange \eqref{eq:bdf_short} and take the
+divergence on both sides
+
+\begin{align}
+    \label{eq:prespois}
+    \nabla^2 p^{n+1} = \nabla \cdot (L_{\times}^\*(u^{n+1} + F^\*(u^{n+1})),
+\end{align}
+
+which is closed by the Neumann type boundary condition
+
+\begin{align}
+    \nabla^2 p^{n+1} \cdot \hat{n} = -\frac{b_0}{\Delta t} \cdot \hat{n}
+    + \nabla \cdot (L_{\times}^\*(u^{n+1} + F^\*(u^{n+1})) \cdot \hat{n}.
+\end{align}
+
+We will refer to this as the pressure Poisson equation in the following. The
+last step is a Helmholtz type equation to solve for the implicit (viscous)
+velocity part which is also derived from \eqref{eq:bdf_short}. Consider
+
+\begin{align}
+    \label{eq:hlm}
+    \frac{b_0}{\Delta t} u^{n+1} - L(u^{n+1}) = -\nabla p^{n+1} + F^*(u^{n+1})
+\end{align}
+
+with the Dirichlet (essential type) boundary condition
+
+\begin{align}
+    u^{n+1} = g_D^{n+1}.
+\end{align}
+
+A detailed walk through can also be found in Franco et al (2020) [^2].
+
+**Note** *The notation is very similar to what is used in the code to make it easy
+to follow the theoretical explanation and understand what is done in the
+implementation.*
+
+[^1]: A. G. Tomboulides, J. C. Y. Lee & S. A. Orszag (1997) Numerical Simulation
+of Low Mach Number Reactive Flows
+
+[^2]: Michael Franco, Jean-Sylvain Camier, Julian Andrej, Will Pazner (2020)
+High-order matrix-free incompressible flow solvers with GPU acceleration and
+low-order refined preconditioners (https://arxiv.org/abs/1910.03032)
+
+## Boundary Conditions
+
+### Inflow and no-slip walls
+
+For inflow or no-slip wall boundary conditions one should use the method
+*NavierSolver::AddVelDirichletBC*. This enforces the value on $u^{n+1}$ in
+\eqref{eq:hlm}. It is valid to call this method multiple times on different
+boundary attributes of the mesh. The *NavierSolver* instance keeps track of the
+associated *Coefficient* and accompanying boundary attribute. The passed
+attribute array can be modified, deleted or reused, since a copy is created.
+
+### Pressure outlet
+
+If an outlet of a domain is supposed to represent a pressure outlet (e.g.
+zero-pressure), one should use the method *NavierSolver::AddPresDirichletBC*.
+This enforces the pressure value $p^{n+1}$ in \eqref{eq:prespois}.
+
+### Zero-stress
+
+This boundary condition is used to represent an outflow attribute. Due to the
+nature of the $H^1$ finite-element discretization, the terms arise naturally in
+\eqref{eq:prespois} and \eqref{eq:hlm} resulting in
+
+\begin{align}
+    \nu \nabla u \cdot \hat{n} - p \cdot \hat{n} = 0.
```

Comment:
Again, p is not a vector quantity so the dot product doesn't seem necessary.

Reply:
Identity tensor was missing.